redis/commands/core.py

```diff
@@ -2454,6 +2454,100 @@ def substr(self, name: KeyT, start: int, end: int = -1) -> ResponseT:
+
+        For more information see https://redis.io/commands/tfunction-load/
+        # TODO: check link when it will be available
+        """
+        pieces: list[EncodableT] = []
+        if replace:
+            pieces.append("REPLACE")
+        if config is not None:
+            pieces.extend(["CONFIG", config])
+        pieces.append(lib_code)
+        return self.execute_command("TFUNCTION LOAD", *pieces)
+
+    def tfunction_delete(self, lib_name: str) -> ResponseT:
+        """
+        Delete a library from RedisGears.
+
+        ``lib_name`` the library name to delete.
+
+        For more information see https://redis.io/commands/tfunction-delete/
+        # TODO: check link when it will be available
+        """
+        return self.execute_command("TFUNCTION DELETE", lib_name)
+
+    def tfunction_list(
+        self, with_code: bool = False, verbose: int = 0, lib_name: str = None
+    ) -> ResponseT:
+        """
+        List the functions with additional information about each function.
+
+        ``with_code`` Show libraries code.
+        ``verbose`` output verbosity level, higher number will increase verbosity level
+        ``lib_name`` specifying a library name (can be used multiple times to show multiple libraries in a single command) # noqa
+
+        For more information see https://redis.io/commands/tfunction-list/
+        # TODO: check link when it will be available
+        """
+        pices: list[EncodableT] = []
+        if with_code:
+            pices.append("WITHCODE")
+        if verbose > 0 and verbose < 4:
+            pices.append("v" * verbose)
+        elif verbose != 0:  # verbose == 0 is the default so no need to throw an error
+            raise DataError("verbose can be 1, 2 or 3")
+        if lib_name is not None:
+            pices.append("LIBRARY")
+            pices.append(lib_name)
+
+        return self.execute_command("TFUNCTION LIST", *pices)
+
+    def tfcall(
+        self,
+        lib_name: str,
+        func_name: str,
+        keys: KeysT = None,
+        *args: List,
+        _async: bool = False,
+    ) -> ResponseT:
+        """
+        Trigger a sync or async (Coroutine) function.
+
+        ``lib_name`` - the library name contains the function.
+        ``func_name`` - the function name to run.
+        ``keys`` - the keys that will be touched by the function.
+        ``args`` - Additional argument to pass to the function.
+        ``_async`` - If True, Invoke an async function (Coroutine.
+
+        For more information see https://redis.io/commands/tfcall/
+        # TODO: check link when it will be available
+        """
+        pieces: list[EncodableT] = [f"{lib_name}.{func_name}"]
+        if keys is not None:
+            pieces.append(len(keys))
+            pieces.extend(keys)
+        else:
+            pieces.append(0)
+        if args is not None:
+            pieces.extend(args)
+        if _async:
+            return self.execute_command("TFCALLASYNC", *pieces)
```

Comment:
@dvora-h any opinion here? I think this would be the first time we did an overload like this - meaning it's now unclear what command is called from the function name. Maybe users would prefer a tfcall_async rather than a boolean.

If so, IMHO we should have both functions call an _tfcall that takes all of the arguments and handles the validation. WDYT?

@uglide Opinion on the API?

Reply:
Agree. I more like the approach of 2 seperate functions, we have it in other places in the library e.g. fcall and fcall_ro

Reply:
@chayim We discussed this when I implemented these commands in [NRedisStack](https://github.com/redis/NRedisStack) and we decided to implement it this way because the two commands have exactly the same input and are pretty much the same command (one is `Invoke a function` and the other is `Invoke an async function (coroutine)`) so you can refer to the command type (`TFCALL` / `TFCALLASYNC`) as an argument.
I can change it but that means I'll have to change it in [NRedisStack](https://github.com/redis/NRedisStack) as well (which would be a Breaking change).

Reply:
I did 2 separate functions